sdk/keyvault/keyvault-keys/src/index.ts

```diff
@@ -681,6 +698,28 @@ export class KeyClient {
     });
   }
 
+  // TODO: is this comment accurate?
+  /**
+   * Rotates the key based on the key policy by generating a new version of the key. This operation requires the keys/rotate permission.
+   *
+   * Example usage:
+   * ```ts
+   * let client = new KeyClient(vaultUrl, credentials);
+   * let key = await client.rotateKey("MyKey");
+   * ```
+   *
+   * @param name - The name of the key to rotate.
+   * @param options - The optional parameters.
+   *
+   * @returns The rotated key.
+   */
+  public rotateKey(name: string, options: RotateKeyOptions = {}): Promise<KeyVaultKey> {
```

Comment:
Add edge case tests - rotate non-existent key, rotate key with and without policy, etc.